### PR TITLE
Edit trail

### PR DESCRIPTION
**Pull request by campercat**
edit-trail feature complete, small fixes on trail form

### CODE REVIEW

File: src/main/frontend/client/components/TrailEdit.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+     name: name,
+     description: description,
+     distance: distance,
+     elevationGain: elevationGain,
+     difficulty: difficulty,
+     zipCode: zipCode,
+     imgUrl: imgUrl
+  });
+
+
```
Comment (safang99) [inline src/main/frontend/client/components/TrailEdit.js:43]: extra empty line we can remove

File: src/main/frontend/client/components/TrailShow.js [hunk 2]
```diff
@@ -46,6 +47,10 @@ const TrailShow = props => {
           <p>Distance: {distance} miles</p>
           <p>Elevation Gain: {elevationGain} ft</p>
           <p>Location: {zipCode}</p>
+          <Link
+            to={{ pathname: `/trails/${trailId}/edit`, state: { trail: trail } }}>
```
Comment (safang99) [inline src/main/frontend/client/components/TrailShow.js:51]: really cool way of passing state!

File: src/main/frontend/client/components/TrailEdit.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  const submitFormHandler = event => {
+    event.preventDefault();
+    setErrors({});
+    updateTrail();
+  };
+
+  if (shouldRedirect) {
+    return <Redirect push to={`/trails/${trailId}`} />;
+  }
+
```
Comment (safang99) [inline src/main/frontend/client/components/TrailEdit.js:84]: This all looks great when a user uses the link to navigate here.  What happens if i type in `/trails/1/edit` and press enter?  Will the form be pre populated with the correct trail information?  We may need to refactor this to send a GET fetch for the form information in a useEffect

**Review by safang99 — CHANGES_REQUESTED** [review]
Thats a really cool way of passing state to the link component! I do want to double check that navigating to the edit page directly is still working ok.  If so everything looks good to me 👍

**Comment by campercat** [comment]
fixed requested changes

**Pull request merged**